application/backend/app/configuration_tools/training_configuration.py

```diff
@@ -0,0 +1,181 @@
+# Copyright (C) 2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from pydantic import BaseModel, Field, model_validator
+
+from .hyperparameters import Hyperparameters
+from .utils import partial_model
+
+
+class SubsetSplit(BaseModel):
+    """
+    Parameters for splitting a dataset into training, validation, and test subsets.
+    The sum of training, validation, and test percentages must equal 100.
+    """
+
+    training: int = Field(
+        ge=1, le=100, default=70, title="Training percentage", description="Percentage of data to use for training"
+    )
+    validation: int = Field(
+        ge=1, le=100, default=20, title="Validation percentage", description="Percentage of data to use for validation"
+    )
+    test: int = Field(
+        ge=1, le=100, default=10, title="Test percentage", description="Percentage of data to use for testing"
+    )
+    auto_selection: bool = Field(
+        default=False, title="Auto selection", description="Whether to automatically select data for each subset"
+    )
+    remixing: bool = Field(default=False, title="Remixing", description="Whether to remix data between subsets")
+    dataset_size: int | None = Field(
+        ge=0,
+        default=None,
+        title="Dataset size",
+        description="Total size of the dataset (read-only parameter, not configurable by users)",
+        exclude=True,  # exclude read-only parameter from serialization
+        json_schema_extra={"default_value": None},
+    )
+
+    @model_validator(mode="after")
+    def validate_subsets(self) -> "SubsetSplit":
+        if (self.training + self.validation + self.test) != 100:
+            raise ValueError("Sum of subsets should be equal to 100")
+        # check that all subsets can have at least one item
+        if self.dataset_size is not None and self.dataset_size < 3:
+            raise ValueError("The dataset is too small to assign at least one item to each subset")
+        return self
+
+
+class MinAnnotationPixels(BaseModel):
+    """Parameters for minimum annotation pixels."""
+
+    enable: bool = Field(
+        default=False,
+        title="Enable minimum annotation pixels filtering",
+        description="Whether to apply minimum annotation pixels filtering",
+    )
+    min_annotation_pixels: int = Field(
```

Comment:
```suggestion
class MinAnnotationPixels(BaseModel):
    """Parameters for minimum annotation pixels."""

    enable: bool = Field(
        default=False,
        title="Enable minimum annotation pixels filtering",
        description="Whether to apply minimum annotation pixels filtering",
    )
    value: int = Field(
```
`Filtering` already has a property `min_annotation_pixels` which maps to this class. Therefore we want this attribute to resolve as `filtering.min_annotation_pixels.value` instead of `filtering.min_annotation_pixels.min_annotation_pixels`.